File: 백준/Silver/10828. 스택/스택.cc
#define _CRT_SECURE_NO_WARNINGS
#include <string.h>
#include <stdio.h>
int top = -1;
int stk[10000];
void push(int item) { stk[++top] = item; }
int pop() { return top < 0 ? -1 : stk[top--]; }
int size() { return top + 1; }
int empty() { return (top == -1) ? 1 : 0; }

int main() {
	char type[20];
	int n, item;
	scanf("%d", &n);
	for (int i = 0; i < n; i++) {
		scanf("%s", type);
		if (strcmp(type, "push") == 0) {
			scanf("%d", &item);
			push(item);
		}
		else if (strcmp(type, "pop") == 0)
			printf("%d\n", pop());
		else if (strcmp(type, "size") == 0)
			printf("%d\n", size());
		else if (strcmp(type, "empty") == 0)
			printf("%d\n", empty());
		else if (strcmp(type, "top") == 0) 
			printf("%d\n", top < 0 ? -1: stk[top]);
	}
	return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.util.Collection;
import java.util.Scanner;
import java.util.Stack;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    Stack<Integer> stack = new Stack<>();

    int n = sc.nextInt();
    for (int i = 0; i < n; i++) {
      String s = sc.next();


      if(s.equals("push")){
        int item = sc.nextInt();
        stack.push(item);
      }
      else if(s.equals("pop")){
        if(stack.isEmpty()){
          System.out.println("-1");
        }
        else{
          System.out.println(stack.pop());
        }

      }

      else if(s.equals("size")){
        System.out.println(stack.size());
      }
      else if(s.equals("empty")){

        System.out.println(stack.isEmpty()?"1":"0");
      }
      else if(s.equals("top")){
        if(stack.isEmpty()){
          System.out.println("-1");
        }
        else{
          System.out.println(stack.peek());
        }
      }

    }


  }
}